packages/google_sign_in/google_sign_in_ios/darwin/Classes/FLTGoogleSignInPlugin.m

```diff
@@ -190,50 +204,65 @@ - (void)requestScopes:(nonnull NSArray<NSString *> *)scopes
   NSSet<NSString *> *requestedScopes = self.requestedScopes;
 
   @try {
-    [self.signIn addScopes:requestedScopes.allObjects
-        presentingViewController:[self topViewController]
-                        callback:^(GIDGoogleUser *addedScopeUser, NSError *addedScopeError) {
-                          BOOL granted = NO;
-                          FlutterError *error = nil;
-                          if ([addedScopeError.domain isEqualToString:kGIDSignInErrorDomain] &&
-                              addedScopeError.code == kGIDSignInErrorCodeNoCurrentUser) {
-                            error = [FlutterError errorWithCode:@"sign_in_required"
-                                                        message:@"No account to grant scopes."
-                                                        details:nil];
-                          } else if ([addedScopeError.domain
-                                         isEqualToString:kGIDSignInErrorDomain] &&
-                                     addedScopeError.code ==
-                                         kGIDSignInErrorCodeScopesAlreadyGranted) {
-                            // Scopes already granted, report success.
-                            granted = YES;
-                          } else if (addedScopeUser == nil) {
-                            granted = NO;
-                          } else {
-                            NSSet<NSString *> *grantedScopes =
-                                [NSSet setWithArray:addedScopeUser.grantedScopes];
-                            granted = [requestedScopes isSubsetOfSet:grantedScopes];
-                          }
-                          completion(error == nil ? @(granted) : nil, error);
-                        }];
+    [self addScopes:requestedScopes.allObjects
```

Comment:
The reason it looks like there are a lot of changes here (and in the previous block) is that switching to the new helper function without `presentingViewController` changed the indentation of the whole block. The code in the block doesn't have any non-whitespace changes.